File: core/src/main/java/inetsoft/web/composer/vs/objects/event/LockVSObjectEvent.java
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package inetsoft.web.composer.vs.objects.event;

import java.io.Serializable;

/**
 * Class that encapsulates the event to change object lock state.
 *
 * @since 12.3
 */
public class LockVSObjectEvent extends VSObjectEvent implements Serializable {
   /**
    * Gets the lock state of object.
    *
    * @return whether object is locked.
    */
   public boolean isLocked() {
      return locked;
   }

   /**
    * Sets the lock state of object.
    *
    * @param locked whether object is locked.
    */
   public void setLocked(boolean locked) {
      this.locked = locked;
   }

   @Override
   public String toString() {
      return "LockVSObjectEvent{" +
         "name='" + this.getName() + '\'' +
         ", isLocked=" + locked +
         '}';
   }

   private boolean locked;
}

File: web/projects/portal/src/app/composer/gui/vs/objects/event/lock-vs-object-event.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { VSObjectEvent } from "../../../../../vsobjects/event/vs-object-event";

/**
 * Event that changes lock status of ve object.
 */
export class LockVSObjectEvent extends VSObjectEvent {
   /**
    * Whether the object is locked.
    */
   public locked: boolean;

   /**
    * Creates a new instance of <tt>MoveVSObjectEvent</tt>.
    *
    * @param objectName the name of the object.
    * @param locked      whether the object is locked.
    */
   constructor(objectName: string, locked: boolean)
   {
      super(objectName);
      this.locked = locked;
   }
}